### ISSUE TITLE
[Bug] [Android] White flash when replacing Current.MainPage with MasterDetailPage

### ISSUE THREAD

**Issue by leo-mck** [xamarin/Xamarin.Forms#8436]
### Description

When replacing the current page with an MasterDetailPage, the screen flashes white between the pages.

### Steps to Reproduce

1.  Create a new Blank XF project, set MainPage.BackgroundColor to something not white
2.  Add an MasterDetailPage using the XF template, that adds an Detail page inside a NavigationPage. 
3. Set the MasterDetailPageDetail.BackgroundCOlor to something not white 
4. On the first page, replace it with the MasterDetailPage = `App.Current.MainPage = new MasterDetailPage1();`

### Expected Behavior
The transition between pages should be seamless
### Actual Behavior
Before the MasterDetailPage appears, there is a white flash
### Basic Information

- Version with issue:  **4.3.0.908675, 4.4.0.936621-pre1**
- Last known good version: 4.2.0.910310
- IDE: VS2019 16.3.7 / VS2019 16.4.0 Preview 3

**Comment by hartez** [comment]
@leo-mck Are you seeing this on a particular platform, or on all of them?

**Comment by leo-mck** [comment]
Sorry, I am seeing this on android.

**Comment by GalaxiaGuy** [comment]
I believe it is the Android window color that shows through, since setting the appropriate Android style like `<item name="android:windowBackground">@color/window_background</item>` "fixes" it. It's only a useful workaround if your pages have the same background color.

**Comment by kerbooo** [comment]
Same issue is for me too.

**Comment by kerbooo** [comment]
Hello all, this bug is fix or not?

**Comment by gisongeorge** [comment]
Any update on this bug? I have the same issue while switching the main page, an intermediate white screen shown in between.

**Comment by GeoSaffer** [comment]
@samhouts is there any feedback on this issue ?

**Comment by samhouts** [comment]
We've fixed this one multiple times. I wonder how it keeps coming back.... https://github.com/xamarin/Xamarin.Forms/pull/3283

**Comment by kerbooo** [comment]
Hello I am not using MDP. This issue does not occurs just MasterDetailPage. I think problem source is NavigationPage.

**Comment by Deijin27** [comment]
> Hello I am not using MDP. This issue does not occurs just MasterDetailPage. I think problem source is NavigationPage. My XF version is 4.5.0.617

@kerbooo 
I had that issue with NavigationPage and realised that it was because I was setting background colors for ContentPages but not NavigationPages. It worked fine once I added to Application.Resources an implicit style

```
<Style TargetType="{x:Type NavigationPage}">
    <Setter Property="BackgroundColor" Value="{DynamicResource AppBackgroundColor}" />
</Style>
```
Maybe that is your issue?

**Comment by actopozipc** [comment]
> 
> 
> I believe it is the Android window color that shows through, since setting the appropriate Android style like `<item name="android:windowBackground">@color/window_background</item>` "fixes" it. It's only a useful workaround if your pages have the same background color.

This fixes it, thanks a lot.

**Comment by Cliffy786** [comment]
> > Hello I am not using MDP. This issue does not occurs just MasterDetailPage. I think problem source is NavigationPage. My XF version is 4.5.0.617
> 
> @kerbooo
> I had that issue with NavigationPage and realised that it was because I was setting background colors for ContentPages but not NavigationPages. It worked fine once I added to Application.Resources an implicit style
> 
> ```
> <Style TargetType="{x:Type NavigationPage}">
>     <Setter Property="BackgroundColor" Value="{DynamicResource AppBackgroundColor}" />
> </Style>
> ```
> 
> Maybe that is your issue?

This helped me, thank you soooooo much, this issue has been a personal headache for **AGES**. you're an absolute life saver! It might  just be a work around, but it definitely makes a huge difference.

**Comment by themronion** [comment]
@Deijin27 GENIUS! Thanks a lot!

**Comment by markelliottopoc** [comment]
Hi,

I'm trying to use this workaround as follows:

<Style TargetType="{x:Type NavigationPage}">
    <Setter Property="BackgroundColor" Value="{StaticResource PrimaryBackground}" />
    <Setter Property="BarBackgroundColor" Value="#ea5c2d" />
</Style>

but am experiencing an exception: "A resource with the key 'Xamarin.Forms.NavigationPage' is already present in the ResourceDictionary".

I haven't added this anywhere else. Any ideas?

**Comment by themronion** [comment]
I am surprised this haven't been fixed. The only possible solution to this problem is to start using Shell, but, for those who are using Prism it is not a valid solution. @jfversluis, maybe, since forms is back on active bug fixing the team can take a look? Happens on all devices starting from android 10.

**Comment by jfversluis** [comment]
Added to the list, thanks!

**Comment by markelliottopoc** [comment]
I have another app that hasn't had the same issue. Not sure what the difference could be though other than I'm targetting a lower minimum android version on the app with the issue. Is there a workaround in the meantime?

**Comment by themronion** [comment]
@markelliottopoc it has something to do with android versions for sure. Since i haven't reproduced it on android 9. The only workaround for me was to switch to Shell

**Comment by markelliottopoc** [comment]
Not an option for me unfortunately. I'm using Prism.

**Comment by themronion** [comment]
@markelliottopoc yeap, starting a new project with Prism and encountered this as well

**Comment by markelliottopoc** [comment]
That's gonna be a fun conversation for us then huh. "Can you change the colour of the nav bar". "Nope!". I have to release the app this week! 

@jfversluis any suggestions?

Would appreciate hugely any guidance.

**Comment by jfversluis** [comment]
@themronion could you add that "new project" with as little extra added as possible that shows the issue? That would be really helpful :)

**Comment by markelliottopoc** [comment]
@jfversluis ok thanks for the swift reply. I've tried everything suggested so far without any luck.

@themronion my project is to big now to send over. I can try and create a new one but I'm actually knee deep in it at the mo moving all my code into a fresh application as the AndroidX migration left my old one unable to build. Would it be straightforward for you to send? Praying it is so!

**Comment by themronion** [comment]
@jfversluis here you are, just a regular app from a Prism template. With a blank XF app i get the same. Also i noticed the flash is much more noticeable in Debug mode than in Release mode, but still it is there. To observe the difference change the ```navigateWithFlash``` variable
[ExpressTaxi.zip](https://github.com/xamarin/Xamarin.Forms/files/7236832/ExpressTaxi.zip)

**Comment by themronion** [comment]
@jfversluis #11993 this is a duplicate of this issue i believe

**Comment by markelliottopoc** [comment]
@jfversluis Just for clarity my issue is that I can't change the nav bar colour. Would you need another issue logged for this?

**Comment by jfversluis** [comment]
@markelliottopoc I noticed. There seems to be a mix here with the two things, that is also why I don't have a clear answer for you. Does #14505 maybe describe your issue better? Are you using Shell? I suspect not as you mention you use Prism?

**Comment by markelliottopoc** [comment]
@jfversluis yes you're right in that this has become a little mixed. My fault. 

I originally came here as I was experiencing the white flash on navigation to a Flyout. I saw the fix stated above:

<Style TargetType="{x:Type NavigationPage}">
    <Setter Property="BackgroundColor" Value="{DynamicResource AppBackgroundColor}" />
</Style>

but got the error: "A resource with the key 'Xamarin.Forms.NavigationPage' is already present in the ResourceDictionary".

I've actually fixed the white flash issue (I think) by improving my code and making sure nothing was running in the main thread when the view is loaded but I still had the problem of not being able to set the navigation bar colour due to the above error.

I've actually just fixed this for all pages **other than** the flyout children (which sit down my navigation stack) with the following set after the initlal navigation in App.xaml.cs:

((NavigationPage)Application.Current.MainPage).BarBackgroundColor = Color.FromHex("#ea5c2d");

But it doesn't appear to work for these flyout detail pages that appear as I navigate through the app.

My navigation stack is something like:

NavigationPage / ViewA / ViewB / ViewC / Flyout / FlyoutDetailA

i.e. ViewA/B/C NavBar colour is set but not FlyoutDetailA.

I've not looked into whether I can fix this yet though. I'll post when I've had a chance to look.

Your tutorial videos have been a big help BTW so thanks ;-) and thanks for taking the time to respond.

**Comment by jfversluis** [comment]
In a related issue someone mentioned that this is resolved in the current Forms version (5.0.0.2337). Could anyone confirm that?

**Comment by jfversluis** [comment]
From some quick testing I think this one might also be resolved as mentioned in #11993 and fixed by #4730

Closing this for now. Let me know if someone thinks there is still an issue here!

**Comment by XamMattia83** [comment]
Hello Gerald i'm Mattia, i had the pleasure to talk with you during your lives...i'm a fan of yours ;)

Well, the problem with the white flash is simply caused by the animation enabled on navigation when you press the back button (hardware button or the navigation bar back arrow). This is very clear if you have a "dark" background.  Users simply need to disable the animation with the navigation APIs; This could be a problem for the hw back button....you have to manually override the back button pressed management to avoid it.
